packages/autorest.go/src/gocodemodel/gocodemodel.ts

```diff
@@ -742,6 +810,10 @@ export function isConstantType(type: PossibleType): type is ConstantType {
   return (<ConstantType>type).values !== undefined;
 }
 
+export function isHeaderMapResponse(resp: HeaderResponse | HeaderMapResponse): resp is HeaderMapResponse {
```

Comment:
I haven't seen this syntax before. I guess the function accepts a parameter of either type but returns undefined when the parameter is a HeaderResponse?

Reply:
[This](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) will explain it better than I can.